src/actions/Upload.ts

```diff
@@ -32,125 +37,126 @@ export interface IReleaseNotes {
  */
 export class Upload extends Edit {
   private uploadParams: IUploadParams
-  private apk: string[]
 
   private versionCodes: any[] = []
 
   constructor (
     client: JWT,
-    apk: string | string[],
-    uploadParams: IUploadParams = {},
+    uploadParams: IUploadParams,
```

Comment:
I think it would be possible to prevent a break of the API by adding this as an optional parameter. 

```ts
// change to Upload class
export class Upload extends Edit {
  private uploadParams: IUploadParams
  private versionCodes: any[] = []
  constructor (
    client: JWT,
    editParams: IEditParams,
    apk?: string | string[], // keep old, but make optional
    uploadParams?: IUploadParams // Add new
  ) {
    super(client, editParams)

    if(!uploadParams) {
      uploadParams = {
        // Turn apk parameter into new format, in case we do not have the new uploadParams parameter.
        files: (typeof apk === 'string' ? [apk] : (apk || [])).map(file => {return {file}})
      }
    }
    assert(
      uploadParams.files.length > 0,
      'I require at least one file to upload file'
    )
//...
```

This might have some consequences to the code around this class though, not sure how difficult it would be to accomodate this.

Reply:
necessary change in `src/index.ts:78`

`- const upload = new Upload(this.client, uploadParams, editParams)`
`+ const upload = new Upload(this.client, editParams, undefined, uploadParams)`

Reply:
I'm actually not too worried about making a breaking change since I already will need to for other v2 features. Combining it with the `editParams` object is a little more concise and consistent anyway. Plus renaming it to `files` is probably good since it could be an APK or an AAB.